### PR TITLE
Basic tests for primary selection device

### PR DESCRIPTION
**Pull request by AlanGriffiths**
These SKIP with miral_wlcs_integration.so
They PASS with the egmde "primary-selection" branch (that currently needs the Mir "primary-selection" branch to build).

Almost certainly we will need more tests, but this is a start...

### CODE REVIEW

**Comment by AlanGriffiths** [comment]
I'd check whether the "zwp_primary_selection_device_manager_v1" interface was "supported", _if I could figure out #107_!

File: include/in_process_server.h [hunk 1]
```diff
@@ -206,6 +207,7 @@ class Client
     wl_shm* shm() const;
     wl_data_device_manager* data_device_manager() const;
     wl_seat* seat() const;
+    struct zwp_primary_selection_device_manager_v1* primary_selection_device_manager() const;
```
Comment (wmww) [inline include/in_process_server.h]: Non-blocking but you don't need the `struct` keyword, and I'm trying to move away from using it in this case.

**Comment by AlanGriffiths** [comment]
bors r=wmww

**Pull request merged**